assignment4/day 5/files/flatten.js

```diff
@@ -0,0 +1,16 @@
+function flatten(unflatObject) {
+  var output = {};
+  for (var key in unflatObject) {
+    if ((typeof unflatObject[key]) == 'object') {
+      var flatObject = flatten(unflatObject[key]);
+      for (var lastObjectKey in flatObject) {
+        output[key + '.' + lastObjectKey] = flatObject[lastObjectKey];
+      }
+    } else {
+      output[key] = unflatObject[key];
+    }
+  }
+  return output;
+}
+
+module.exports.flatten = flatten;
```

Comment:
Give a specific reason on why u are following 2 patterns in terms of module.exports